### PR TITLE
docs: update supported Go information and test config

### PR DESCRIPTION
**Pull request by zlesnr**
fixes: #474

### CODE REVIEW

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -12,8 +12,8 @@ jobs:
     strategy:
       matrix:
         go-version:
-           - 1.13.x
            - 1.14.x
+           - 1.15.x
```
Comment (sanderblue) [inline .github/workflows/compile.yml:16]: Wondering if we should also keep `1.13.x` around just so we're informed if we introduce a breaking change for an older version of Go. Thoughts?

Comment (zlesnr) [reply .github/workflows/compile.yml:16]: If we did and the build failed, would that prevent us from merging the PR?

Comment (sanderblue) [reply .github/workflows/compile.yml:16]: Yeah, but likely won't run into that problem. And when we do, we can make a decision on whether or not to remove that version of the build matrix, or update the code in question, or create a separate parallel job that runs on Go 1.13 that's allowed to fail. 

Also totally fine keeping things as they are in this PR. I'm more so thinking about making sure we stay informed when/if we introduce something messes with things.

Comment (zlesnr) [reply .github/workflows/compile.yml:17]: Okay, I've added 1.13 back in.  You make a good point.

**Review by sanderblue — APPROVED** [review]
🐑 it

**Review by sanderblue — APPROVED** [review]
🐑 it

**Pull request merged**